### PR TITLE
fix: inflate crash and media provider bug

### LINKED ISSUE #2927: Building App with SDK 29 Crash on Android 10
**Describe the bug**
Building the App with SDK Tools 29 and BuildTools 29.0.2 will not run on android 10 devices.
The official version of the app i.e "1.10.0" is also built with SDK 29 ,
Hence the app is crashing on all android 10 devices.

**To Reproduce**
1. Use any android 10 device or emulator
2. Install from playStore or run from debug
3. App Crash on SplashScreen

**Expected behavior**
Should Run on Android 10 devices.

**Logs**
```
2019-11-19 05:33:57.502 15158-15158/org.fossasia.phimpme E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.fossasia.phimpme, PID: 15158
    java.lang.RuntimeException: Unable to start activity ComponentInfo{org.fossasia.phimpme/org.fossasia.phimpme.gallery.activities.SplashScreen}: android.view.InflateException: Binary XML file line #17 in org.fossasia.phimpme:layout/abc_screen_simple: Binary XML file line #17 in org.fossasia.phimpme:layout/abc_screen_simple: Error inflating class androidx.appcompat.widget.FitWindowsLinearLayout
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3374)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3513)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2109)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7682)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:516)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
     Caused by: android.view.InflateException: Binary XML file line #17 in org.fossasia.phimpme:layout/abc_screen_simple: Binary XML file line #17 in org.fossasia.phimpme:layout/abc_screen_simple: Error inflating class androidx.appcompat.widget.FitWindowsLinearLayout
     Caused by: android.view.InflateException: Binary XML file line #17 in org.fossasia.phimpme:layout/abc_screen_simple: Error inflating class androidx.appcompat.widget.FitWindowsLinearLayout
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object java.lang.reflect.Field.get(java.lang.Object)' on a null object reference
        at com.mikepenz.iconics.context.ReflectionUtils.getValue(ReflectionUtils.java:29)
        at com.mikepenz.iconics.context.InternalLayoutInflater.createCustomViewInternal(InternalLayoutInflater.java:155)
        at com.mikepenz.iconics.context.InternalLayoutInflater.access$000(InternalLayoutInflater.java:20)
        at com.mikepenz.iconics.context.InternalLayoutInflater$PrivateWrapperFactory2.onCreateView(InternalLayoutInflater.java:246)
        at android.view.LayoutInflater$FactoryMerger.onCreateView(LayoutInflater.java:239)
        at android.view.LayoutInflater.tryCreateView(LayoutInflater.java:1069)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:997)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:961)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:659)
        at com.mikepenz.iconics.context.InternalLayoutInflater.inflate(InternalLayoutInflater.java:81)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:534)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:481)
        at androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor(AppCompatDelegateImpl.java:773)
        at androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor(AppCompatDelegateImpl.java:659)
        at androidx.appcompat.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:552)
        at androidx.appcompat.app.AppCompatActivity.setContentView(AppCompatActivity.java:161)
        at org.fossasia.phimpme.base.BaseActivity.onCreate(BaseActivity.java:53)
        at org.fossasia.phimpme.base.ThemedActivity.onCreate(ThemedActivity.java:40)
        at org.fossasia.phimpme.base.SharedMediaActivity.onCreate(SharedMediaActivity.java:12)
        at org.fossasia.phimpme.gallery.activities.SplashScreen.onCreate(SplashScreen.java:63)
        at android.app.Activity.performCreate(Activity.java:7815)
        at android.app.Activity.performCreate(Activity.java:7804)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1318)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3349)
2019-11-19 05:33:57.502 15158-15158/org.fossasia.phimpme E/AndroidRuntime:     at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3513)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2109)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7682)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:516)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
```

**GIF Showing Crash of Debug Release using SDK 29**
ezgif com-video-to-gif

**GIF Showing Crash of latest version App from PlayStore**

ezgif com-video-to-gif (1)

**Smartphone Info:**

|               | |
|---------------|-|
|Device         |OnePlus7|
|Android Version|Android 10 OxygenOS 10.0.2    |

**Additional context**
The App runs fine on android versions < 10.
It seems to occur because of some outdated dependency libraries.
Since on android sdk 29, androidx has components been made mandatory ,
It might be that one of the library most probably [Android-Iconics](https://github.com/mikepenz/Android-Iconics) is causing the issue ( As checked from the logs ).
Currently the fast work-ground can be to use SDK 28 to build the project.

**Would you like to work on the issue?**

- [ ] Yes
- [ ] No
- Other: 
For fast solution I can create a PR to downgrade the sdk to 28 ,
Need your suggestions and reviews.

### LINKED ISSUE #2982: Crash on application start
**Describe the bug**
I forked [phimpme-android](https://github.com/fossasia/phimpme-android) development branch to fix it because on my device it crashes on startup. The fix was migrating to iconics 4.0.2
The migrated app can be found here: [phimpme fork](https://github.com/TBog/phimpme-android/commits/tbog/iconics)
Now it still crashes on app start-up inside `gallery\data\providers\MediaStoreProvider.java` in method 
`public static ArrayList<Album> getAlbums(Context context)`

**To Reproduce**
Upon app start the app closes instantly.
Seems like the query has changed and the GROUP BY 'hack' used is not longer working.

**Logs**
```
2020-01-17 10:13:30.096 9755-10705/? E/DatabaseUtils: Writing exception to parcel
    android.database.sqlite.SQLiteException: near "GROUP": syntax error (code 1 SQLITE_ERROR[1]): , while compiling: SELECT parent, bucket_display_name FROM files WHERE ((_data>'') AND (owner_package_name IN ( 'org.fossasia.phimpme' ) OR media_type=2 OR media_type=4 OR media_type=0 AND mime_type LIKE 'audio/%' OR media_type=3 OR media_type=0 AND mime_type LIKE 'video/%' OR media_type=1 OR media_type=0 AND mime_type LIKE 'image/%') AND (is_pending=0) AND (is_trashed=0) AND (volume_name IN ( 'external_primary' ))) AND ((media_type=? ) GROUP BY ( parent ))
        at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
        at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:1364)
        at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:803)
        at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:590)
        at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:62)
        at android.database.sqlite.SQLiteQuery.<init>(SQLiteQuery.java:37)
        at android.database.sqlite.SQLiteDirectCursorDriver.query(SQLiteDirectCursorDriver.java:46)
        at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1959)
        at android.database.sqlite.SQLiteQueryBuilder.query(SQLiteQueryBuilder.java:604)
        at com.android.providers.media.MediaProvider.queryInternal(MediaProvider.java:2167)
        at com.android.providers.media.MediaProvider.query(MediaProvider.java:1978)
        at android.content.ContentProvider$Transport.query(ContentProvider.java:267)
        at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:104)
        at android.os.Binder.execTransactInternal(Binder.java:1021)
        at android.os.Binder.execTransact(Binder.java:994)
```

**Smartphone Info:**

|               | |
|---------------|-|
|Device         | Samsung Galaxy S10 plus |
|Android Version| Android version 10    |

**Would you like to work on the issue?**

- [x] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by TBog**
fixes inflate crash by removing `IconicsContextWrapper`, it's deprecated anyway
Also fixes #2982
 fixes #2927

### CODE REVIEW

**Comment by TBog** [comment]
@codedsun please review this

**Comment by codedsun** [comment]
@TBog - Why you closed the previous PR and also is this working on android 10?

**Comment by TBog** [comment]
In the previous PR I thought that using the latest iconics library was the solution, but I found out it's only fixing it on Android 10. This PR fixes the problem on both Android 10 and 9.
For Android 10 I also cherry picked the query hack fix and the Glide workaround

**Comment by TBog** [comment]
@codedsun please post logs with the errors, i have none.

**Comment by TBog** [comment]
@codedsun Please help me merge this into master! I have this working on Android 9 on the emulator and it's working on Android 10 on my SGS10+

File: app/src/main/java/org/fossasia/phimpme/gallery/data/providers/MediaStoreProvider.java [hunk 1]
```diff
@@ -85,14 +85,10 @@ private static boolean isExcluded(String path, Context context) {
...
-    selection =
-        MediaStore.Files.FileColumns.MEDIA_TYPE
-            + "=? ) GROUP BY ( "
-            + MediaStore.Files.FileColumns.PARENT
-            + " ";
-    selectionArgs = new String[] {String.valueOf(MediaStore.Files.FileColumns.MEDIA_TYPE_IMAGE)};
+    String selection = MediaStore.Files.FileColumns.MEDIA_TYPE + "=? ";
+    String[] selectionArgs =
+        new String[] {String.valueOf(MediaStore.Files.FileColumns.MEDIA_TYPE_IMAGE)};
+    String sortOrder = MediaStore.Files.FileColumns.PARENT;
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/phimpme/gallery/data/providers/MediaStoreProvider.java:91]: Why changes in this file?

File: app/src/main/java/org/fossasia/phimpme/gallery/data/providers/MediaStoreProvider.java [hunk 2]
```diff
@@ -102,23 +98,26 @@ private static boolean isExcluded(String path, Context context) {
...
+        HashSet<Long> idSet = new HashSet<>();
         do {
           Media media = getLastMedia(context, cur.getLong(idColumn));
           if (media != null && media.getPath() != null) {
             String path = StringUtils.getBucketPathByImagePath(media.getPath());
             boolean excluded = isExcluded(path, context);
-            if (!excluded) {
+            long curIdColumn = cur.getLong(idColumn);
+            if (!excluded && !idSet.contains(curIdColumn)) {
+              idSet.add(curIdColumn);
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/phimpme/gallery/data/providers/MediaStoreProvider.java:115]: Why changes in this file?

**Pull request merged**